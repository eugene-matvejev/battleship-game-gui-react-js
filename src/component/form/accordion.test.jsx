import React from 'react';
import PropTypes from 'prop-types';

const Accordion = ({ qaSelector, className, title, onCollapse, isCollapsed, children, ...props }) =>
    <section data-cy={qaSelector} className={`accordion ${className}`} {...props}>
        {title && <h3 onClick={onCollapse}>{title}</h3>}
        {!isCollapsed && children}
    </section>;

Accordion.propTypes = {
    qaSelector: PropTypes.string,
    className: PropTypes.string,
    title: PropTypes.string,
    onCollapsed: PropTypes.func,
    isCollapsed: PropTypes.bool,
};
Accordion.defaultProps = {
    qaSelector: '',
    className: '',
};

export default Accordion;
