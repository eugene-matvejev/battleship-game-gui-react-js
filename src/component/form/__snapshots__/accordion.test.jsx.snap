// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion/> render with default/required props 1`] = `
<section
  className="accordion "
/>
`;
